cpp/arcticdb/version/version_utils.hpp

```diff
@@ -217,12 +217,22 @@ inline bool load_latest_ongoing(const LoadParameter& load_params, const std::sha
     return false;
 }
 
-inline bool looking_for_undeleted(const LoadParameter& load_params, const std::shared_ptr<VersionMapEntry>& entry) {
-    if(!(load_params.load_type_ == LoadType::LOAD_UNDELETED && entry->tombstone_all_))
+inline bool looking_for_undeleted(const LoadParameter& load_params, const std::shared_ptr<VersionMapEntry>& entry, const VersionId& oldest_loaded_index) {
+    if(load_params.load_type_ != LoadType::LOAD_UNDELETED) {
         return true;
+    }
 
-    ARCTICDB_DEBUG(log::version(), "Exiting because we have found an undeleted version");
-    return false;
+    if(entry->tombstone_all_) {
+        const bool is_deleted_by_tombstone_all = entry->tombstone_all_->version_id() >= oldest_loaded_index;
+        if(is_deleted_by_tombstone_all) {
+            ARCTICDB_DEBUG(log::version(), "Exiting because we have found an undeleted version");
```

Comment:
It's very annoying that `LoadType` is undocumented. My understanding of `LOAD_UNDELETED` is that it should load *all* undeleted versions.

That means the logging message is misleading: we did not exit because "an undeleted version" was found (that's the job of `LOAD_LATEST_UNDELETED`), but because we found a tombstone all key, so there couldn't be any further undeleted versions.

Reply:
I was about to ask about this as well. It doesn't make much sense the way it is now. Maybe it should be `Exiting because tombstone_all key deletes all versions beyond version ...`. What do you think?